packages/plugin-nested-docs/src/hooks/resaveChildren.ts

```diff
@@ -36,51 +42,67 @@ const resave = async ({ collection, doc, draft, pluginConfig, req }: ResaveArgs)
     },
   })
 
-  const breadcrumbSlug = pluginConfig.breadcrumbsFieldSlug || 'breadcrumbs'
-
-  try {
-    await children.docs.reduce(async (priorSave, child) => {
-      await priorSave
+  const publishedChildren = await req.payload.find({
+    collection: collection.slug,
+    depth: 0,
+    draft: false,
+    locale: req.locale,
+    req,
+    where: {
+      [parentSlug]: {
+        equals: doc.id,
+      },
+    },
+  })
 
-      const childIsPublished =
-        typeof collection.versions === 'object' &&
-        collection.versions.drafts &&
-        child._status === 'published'
+  const childrenById = [...draftChildren.docs, ...publishedChildren.docs].reduce((acc, child) => {
+    acc[child.id] = acc[child.id] || []
+    acc[child.id].push(child)
+    return acc
+  }, {})
 
-      if (!parentDocIsPublished && childIsPublished) {
-        return
+  const sortedChildren = Object.values(childrenById).flatMap((group: JsonObject[]) => {
+    return group.sort((a, b) => {
+      if (a.updatedAt !== b.updatedAt) {
+        return a.updatedAt > b.updatedAt ? -1 : 1
       }
+      return a._status === 'published' ? -1 : 1
+    })
+  })
 
-      await req.payload.update({
-        id: child.id,
-        collection: collection.slug,
-        data: {
-          ...child,
-          [breadcrumbSlug]: await populateBreadcrumbs(req, pluginConfig, collection, child),
-        },
-        depth: 0,
-        draft: !childIsPublished,
-        locale: req.locale,
-        req,
-      })
-    }, Promise.resolve())
-  } catch (err: unknown) {
-    req.payload.logger.error(
-      `Nested Docs plugin has had an error while re-saving a child document${
-        draft ? ' as draft' : ' as published'
-      }.`,
-    )
-    req.payload.logger.error(err)
+  if (sortedChildren) {
```

Comment:
This code seems fine, but what is the reason for sorting all draft and published docs by updatedAt?

Reply:
Before sorting, the array has the published docs first even when the latest doc is a draft, which ends up making any draft children docs published - I needed to sort by `updatedAt` to update them in the correct order.